FrameworkSystem/scripts/dirac-admin-update-instance.py

```diff
@@ -0,0 +1,248 @@
+    for host in updateFail:
+      gLogger.notice(" - %s" % host)
+    gLogger.notice("X!X!X!X!X!X!X!X!X!X!X!X!X!X!X!X!X!X!")
+    return S_OK([updateSuccess, updateFail])
+
+
+def restartHost(hostName):
+  """
+  Restart all systems and components of a host
+
+  :params hostName: name of the host you want to restart
+  """
+  hostList = hostName.split(':')
+  host = hostList[0]
+  if len(hostList) == 2:
+    port = hostList[1]
+  else:
+    port = None
+
+  gLogger.notice("Pinging %s ..." % host)
+
+  client = SystemAdministratorClient(host, port)
+  result = client.ping()
+  if result['OK']:
+    gLogger.notice("Host %s is active" % host)
+  else:
+    gLogger.error("Could not connect to %s: %s" % (host, result['Message']))
+    return result
+
+  gLogger.notice("Initiating restart of all systems and components")
+  result = client.restartComponent('*', '*')
+  gLogger.notice("All systems are restarted, connection to SystemAdministrator is lost:" + result['Message'])
+  return S_OK(result['Message'])
+
+
+def updateInstance(version):
+  """
+  Update each server of an instance and restart them
+
+  :params version: version vArBpC you want to update to
+  """
+  result = updateHosts(version)
+  if not result['OK']:
+    return result
+
+  updateSuccess = result['Value'][0]
+  updateFail = result['Value'][1]
+  restartSuccess = []
+  restartFail = []
+  for host in updateSuccess:
+    result = restartHost(host)
+    if result['OK']:
+      restartSuccess.append(host)
+    else:
+      restartFail.append(host)
+
+  if not restartFail and not updateFail:
+    return S_OK("Successfully updated and restarted all hosts")
+  else:
+    gLogger.notice("XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX")
+    gLogger.notice("XXXXX There were problems in the update process XXXXX")
+    gLogger.notice("Succeeded to update:")
+    for host in updateSuccess:
+      gLogger.notice(" + %s" % host)
+    gLogger.notice("Succeeded to restart:")
+    for host in restartSuccess:
+      gLogger.notice(" + %s" % host)
+    gLogger.notice("Failed to update:")
+    for host in updateFail:
+      gLogger.notice(" - %s" % host)
+    gLogger.notice("Failed to restart:")
+    for host in updateFail:
+      gLogger.notice(" - %s" % host)
+    gLogger.notice("XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX")
+    return S_ERROR("Update failed!")
+
+
+def updatePilot(version, vo):
+  """
+  Update in the CS the pilot verion used,
+  If only one version present in CS it's overwritten.
+  If two versions present, the new one is added and the last removed
+
+  :params version: version vArBpC of pilot you want to use
+  :params vo: Location of pilot version in CS /Operations/<vo>/Pilot/Version
+  """
+  setup = vo
+  if not vo:
+    setup = gConfig.getValue('/DIRAC/DefaultSetup')
+  if not setup:
+    return S_ERROR("No value set for /DIRAC/DefaultSetup in CS")
+
+  pilotVersion = gConfig.getValue('Operations/%s/Pilot/Version' % setup)
```

Comment:
```suggestion
  pilotVersion = gConfig.getValue('Operations/%s/Pilot/Version' % setup, [])
```
then you already get a list.